frontend/src/components/MonacoEditor.tsx

```diff
@@ -17,7 +18,9 @@ interface Getter {
 
 const MonacoEditor = ({ content, filename }: Props) => {
   const [dialogOpen, setDialogOpen] = useState(false)
-  const currentBranch = 'master' // get this from somewhere
+
+  const branchState = useQuery<RepoStateQueryResult>(BRANCH_STATE)
+  const currentBranch = branchState.data?.repoState.branchName || ''
```

Comment:
If `branchState.data` is null or undefined, the `currentBranch` would just be an empty string. I'm not sure if that's acceptable. Otherwise looks good.

Reply:
Suggested fixes in daily: Disable save button while `branchState.loading` is `true`.